editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/submissionDataTypesAndQueries.ts

```diff
@@ -0,0 +1,75 @@
+import gql from "graphql-tag";
+import { client } from "lib/graphql";
+
+type PaymentRequest = {
+  id: string;
+  createdAt: string;
+  paidAt: string;
+  govPaymentId: string;
+};
+
+type PaymentStatus = {
+  govPaymentId: string;
+  createdAt: string;
+  status: string;
+};
+
+type BopsApplication = {
+  id: string;
+  submittedAt: string;
+  destinationUrl: string;
+};
+
+type EmailApplication = {
+  id: string;
+  recipient: string;
+  submittedAt: string;
+};
+
+type UniformApplication = {
+  id: string;
+  submittedAt: string;
+};
+
+export type SubmissionData = {
```

Comment:
nit: This type is just used in the component as part for the `useState()` hook. It would be better if also tied to `fetchSubmittedApplications()` which in future could be used outside of this context.

Two options here - 
 - assign a return type to the function (e.g. `Promise<SubmissionData>`)
 - Pass `SubmissionData` in as a generic type to `client.query()` (e.g. `await client.query<SubmissionData>(...)`

Reply:
This is a great suggestion! This didn't occur to me but is such a nice way to tighten up the types 🙌 

Fixed here: cc2eb174e138a23046a710c38da83bdcc9d63a54